src/molecule/dependency/base.py

```diff
@@ -80,13 +82,32 @@ def execute_with_retries(self):
         LOG.error(str(exception), self._sh_command)
         util.sysexit(getattr(exception, "exit_code", constants.RC_UNKNOWN_ERROR))
 
-    @abc.abstractmethod
-    def execute(self):  # pragma: no cover
+    def execute(self):
         """
         Execute ``cmd`` and returns None.
 
         :return: None
         """
+        if os.path.isfile("galaxy.yml"):
+            LOG.info("Collection detected at %s", os.getcwd())
+
+            dist = Path(self._config.scenario.ephemeral_directory) / "dist"
+            dist.mkdir(parents=True, exist_ok=True)
+            result = util.run_command(
+                f"ansible-galaxy collection build -v -f --output-path {dist}",
```

Comment:
Again, this build command is too naive because it assumes that we are running molecule from the collection root. And there is another problem: users might need to some something else before they build the collection artifact. Because `ansible-galaxy collection build` command is really primitive, there is usually quite a few steps we need to perform before we get to building (or we risk packaging together things like virtual environments, coverage reports - and the are huge BTW, etc.).